src/lib/payload.ts

```diff
@@ -15,7 +15,44 @@ export const demandCreate = (demand: DemandPayload, transactionId: UUID) => ({
         state: { type: 'LITERAL', value: DemandState.created },
         subtype: { type: 'LITERAL', value: demand.subtype },
         parameters: { type: 'FILE', value: demand.filename },
-        transactionId: { type: 'LITERAL', value: transactionId.replace(/[-]/g, '') },
+      },
+    },
+  ],
+})
+
+export const match2Propose = (match2: Match2Response, demandA: DemandPayload, demandB: DemandPayload) => ({
+  files: [],
+  process: { id: 'match2-propose', version: 1 },
+  inputs: [demandA.latestTokenId, demandB.latestTokenId],
+  outputs: [
+    {
+      roles: { Owner: demandA.owner },
+      metadata: {
+        version: { type: 'LITERAL', value: '1' },
+        type: { type: 'LITERAL', value: TokenType.DEMAND },
+        state: { type: 'LITERAL', value: DemandState.created },
+        subtype: { type: 'LITERAL', value: demandA.subtype },
+        originalId: { type: 'TOKEN_ID', value: demandA.originalTokenId },
+      },
+    },
+    {
+      roles: { Owner: demandB.owner },
+      metadata: {
+        version: { type: 'LITERAL', value: '1' },
+        type: { type: 'LITERAL', value: TokenType.DEMAND },
+        state: { type: 'LITERAL', value: DemandState.created },
+        subtype: { type: 'LITERAL', value: demandB.subtype },
+        originalId: { type: 'TOKEN_ID', value: demandB.originalTokenId },
+      },
+    },
+    {
+      roles: { Optimiser: match2.optimiser, MemberA: match2.memberA, MemberB: match2.memberB },
```

Comment:
I've found myself using `DemandA.owner` and `DemandB.owner` instead of the roles. Are the roles perhaps redundant? Or maybe I should be checking against them